src/Engine/IO/Window.cpp

```diff
@@ -117,14 +150,28 @@ void Window::update(float dt) {
     Mouse::update(x, y);
 
     /* Update ImGui */
+    // TODO: clean up this code
+    static bool imGuiActive = false;
+    static bool priorCursorState;
     imGuiTimer += dt;
-    if (Keyboard::isKeyPressed(GLFW_KEY_GRAVE_ACCENT) && 
-       (Keyboard::isKeyPressed(GLFW_KEY_LEFT_SHIFT) || Keyboard::isKeyPressed(GLFW_KEY_RIGHT_SHIFT)) &&
-        imGuiTimer >= 0.5) {
+    if (Keyboard::isKeyPressed(GLFW_KEY_GRAVE_ACCENT) && imGuiTimer >= 0.5) {
+        if (!imGuiActive) {
+            priorCursorState = cursorEnabled;
+            setCursorEnabled(true);
+        }
+
         toggleImGui();
         imGuiTimer = 0.0;
+        imGuiActive = !imGuiActive;
```

Comment:
How is this different than toggieImGui()

Reply:
it's not. i didn't realize that's all toggleImGui did. Removed imGuiActive